_posts/2019-01-02-Deviceid-Reliability.md

```diff
@@ -0,0 +1,95 @@
+---
+layout: post
+title: Keep the quality of your data high
+subtitle: All that glitters is not gold
+category: data
+tags: [adtech, howto]
+author: Alessio Borgheresi
+header-img: "images/Deviceid-Reliability/data_quality.jpg"
+---
+
+This post describes an example of the quality selection we perform at Adello, underlining the benefits we gain from it.
+
+In a [previous post](https://adello.github.io/Tracking/) we introduced the way we track a person (a smart device) for advertising targeting purpose.
+In summary, we use the Android **[Advertising ID](https://support.google.com/googleplay/android-developer/answer/6048248?hl=en)** and the iOS Identifier for Advertisers. 
+In the rest of the post we call these random pseudonyms as **deviceID** since we use them as an identifier of a smart device (mobile, pad, etc...).
+
+It is expected that:
+- behind a deviceID there should be a real physical device used by a human person, 
+- when the a physical device uses different apps, those should give the same deviceID (unless it is resetted by the device's user which definetively does not occur at a per minute/hour frequency).
+
+Here we want to discover the deviceID which looks like real data but after some statistical studies you can categorize them as non reliable/useful in the sense that it is not guarantee that a real person is using the device.
+Dealing with null missing values or out-of-range values it is not what we are talking about.
+
+The data can be described as a collection of deviceIDs which visit an app at a certain time.
+<p align="center"> <img src="../images/Deviceid-Reliability/table_data_example.png"> </p>
+
+
+Let's go into details of the discovered patterns:
+
+* **Fraction of unique devices**<br/>
+Only apps with a certain number of distinct deviceIDs must be considered to avoid statistical fluctuations.
+With the histogram of the fraction of unique devices we can see both apps which are visited by too many/few distinct deviceIDs as also apps which shows some regular traffic.
+![frac_uniq_dev](../images/Deviceid-Reliability/fraction_unique_devices.png)
+The peaks in at one are those which are visited by always distinct deviceid while the other agglomerate close to zero.
+
+We suspect that something is broken in the way the apps provide/access the deviceID information for those apps.
+All deviceIDs which appear in a non reliable flagged app are considered non reliable, a simple inner join over the selected apps is enought to create the blacklist.
+
+
+* **Non realistic number of apps**<br/>
+For a fixed amount of time (e.g. 1 day) it is possible to observe the number of apps visited by each deviceID.
+From this plot we found out some clear outliers which cannot correspond to real human users.
```

Comment:
What are those outliers? Explain explicitly: We see, that there are device identifiers, which can be seen on more than 500 different apps during one day.